lib/klarna_gateway/models/discount_item_serializer.rb

```diff
@@ -13,8 +13,8 @@ def to_hash
         # send the name and the promo code
         name: name.presence || "Discount",
         reference: "Discount",
-        total_amount: (@order.promo_total * 100).to_i,
-        unit_price: (@order.promo_total * 100).to_i,
+        total_amount: (@order.promo_total * 100 + @order.total_applicable_store_credit * 100).to_i,
```

Comment:
I'm looking at `total_applicable_store_credit` and noticing how it changes depending on the state of the order. I think either way is OK but do you know at what order `state`/`states` this code gets invoked?